### ISSUE TITLE
unable to run the mtcar example

### ISSUE THREAD

**Issue by windwine** [matloff/polyreg#7]
I was trying to run the example of

y <- mtcars[,1]
data <- cbind(mtcars[,-1], y) # make y column the last column
f <- polyFit(data,2,2,"lm",TRUE, 0.9)
pred <- predict(f,data[,-ncol(data)])

after installing following

install.packages("devtools")
library(devtools)
install_github("matloff/polyreg")

I got error messages of
"Error in UseMethod("predict") : 
  no applicable method for 'predict' applied to an object of class "eigen"".

Did I got the package installation wrong?

**Comment by matloff** [comment]
Problem on our end, working on it. Run this example without the PCA option for now. (PCA works fine in general, but the small size of this problem is causing an anomaly.)

**Comment by windwine** [comment]
Hi Prof. Matloff,

Thank you for your replies. I have tried another problem at hand with 94 predictors and 300K observations. The fit used 8 principal components and I still have the the same error messages. Meanwhile, I am also thinking about another expansion to the polyreg following the random forest idea. Since polyreg can run very fast as it is only dealing with a few principal components we would be able to use RF's setup to do random-polyreg especially when dealing with noisy data. If we view regression tree as another ployreg with all categorical Xs I guess the benefit from RF could also apply to random-polyreg. It would be impractical to run random-DNN as DNN is way more computational intensive compared to tree or polyreg. BTW, I am a big fan of your book and thank you.

Regards,

Jiaqi

**Comment by matloff** [comment]
Thanks for the suggestion and nice comment. Would it be possible to obtain your dataset and code, so we can replicate the problem and fix it?

**Comment by windwine** [comment]
I have saved the first 400 observations into an rds file and it has the same error message as I ran the whole 300K observations. The last column is y and the second to last column is the monthly indicator so its value is 1,2,...,12. I did not set this column to be categorical and will that be the problem? Thanks.

[data.zip](https://github.com/matloff/polyreg/files/2184701/data.zip)


data1<-readRDS("data.rds")
f <- polyFit(data1[1:300,],2,2,"lm",TRUE, 0.99)


pred <- predict(f,data1[301:400,-ncol(data1)])
